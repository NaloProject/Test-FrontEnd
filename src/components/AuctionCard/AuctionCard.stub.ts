import { Sale } from '@Redux/types'

export const auctionCardStub: Sale = {
  id: '6',
  name: 'Ceres Searching',
  totalMinted: 2,
  price: '10.89',
  like: 400,
  forSale: 1,
  timeLeft: '2 days',
  isHot: true,
  sellerId: 2323,
  image: '/images/ntfs/Contemporary-Sculptures-For-The-Home.jpg',
}
