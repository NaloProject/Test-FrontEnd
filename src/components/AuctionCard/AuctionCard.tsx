import { FC } from 'react'
import { Sale } from '@Redux/types'
import { selectSellerById } from '@Redux/nfts/selectors'
import { useAppSelector } from '@Redux/hooks'
import { ReactComponent as FlameIcon } from '@Svg/flame-icon.svg'
import { ReactComponent as FavoriteIcon } from '@Svg/favorite.svg'
import Image from 'next/image'
import Link from 'next/link'
import SellerCard from '@Components/SellerCard'

export type AuctionCardProps = {
  auction: Sale
}

const AuctionCard: FC<AuctionCardProps> = ({ auction }) => {
  const sellerById = useAppSelector((state) =>
    selectSellerById(state, auction.sellerId)
  )

  return (
    <article className="flex flex-col rounded-xl mr-1 sm:mr-2 md:mr-4 h-full">
      <div className="flex flex-col w-full p-[16px] border border-slate-200 dark:border-slate-600 rounded-tl-xl rounded-tr-xl min-h-[400px]">
        <header className="w-full relative mb-4">
          <div className="overflow-hidden flex rounded-lg w-full relative w-full h-full !relative">
            <Link
              href={`/auction/${auction.id}`}
              key={auction.id}
              className="w-full h-full !relative"
            >
              <Image
                src={auction.image}
                alt={auction.name}
                fill
                className="pointer-events-none object-cover !h-auto !relative"
              />
            </Link>
          </div>
          <div className="absolute bg-white bottom-[20px] flex left-[20px] rounded-lg px-3 py-2">
            <span className="text-slate-700 font-medium mr-2 text-[12px]">
              {auction.timeLeft}
            </span>
            {auction.isHot ? (
              <span className="inline-block w-[12px] h-[12px]">
                <FlameIcon />
              </span>
            ) : null}
          </div>
        </header>
        <div className="flex mb-3">
          <div className="flex-1">
            <h3 className="font-medium text-slate-700 dark:text-slate-300 mb-1">
              {auction.name}
            </h3>
            <h4 className="text-[12px] text-slate-300 dark:text-slate-600">
              {auction.totalMinted} Editions minted
            </h4>
          </div>
          <div className="text-[13px] text-green-400">
            {auction.forSale} for sale
          </div>
        </div>
        {sellerById && (
          <SellerCard className="mt-auto" seller={sellerById} size="small" />
        )}
      </div>
      <footer className="flex p-[16px] border border-slate-200 dark:border-slate-600 border-t-0 min-h-[77px] items-center justify-between rounded-bl-xl rounded-br-xl w-full">
        <div className="flex flex-col">
          <span className="font-bold text-[14px] text-slate-700 dark:text-slate-300 mb-1">
            {auction.price} ETH
          </span>
          <span className="text-[13px] text-slate-400">Starting Bid</span>
        </div>
        <div className="flex items-center">
          <span className="font-medium text-[11px] text-slate-400 mr-1">
            {auction.like}
          </span>
          <span className="inline-block w-[22px] h-[22px]">
            <FavoriteIcon className="fill-slate-400" />
          </span>
        </div>
      </footer>
    </article>
  )
}

export default AuctionCard
