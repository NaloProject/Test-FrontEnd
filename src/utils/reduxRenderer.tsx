import { configureStore, EnhancedStore } from '@reduxjs/toolkit'
import { render, RenderOptions, RenderResult } from '@testing-library/react'
import { ReactElement, FC, PropsWithChildren } from 'react'
import { Provider } from 'react-redux'

import { combined, RootState } from '@Redux/store'
import INITIAL_STATE from './initialState'

type RenderWithReduxOptions = Omit<RenderOptions, 'queries'> & {
  initialState?: RootState
  store?: EnhancedStore
}

export const renderWithRedux = (
  ui: ReactElement,
  {
    initialState = INITIAL_STATE,
    store = configureStore({
      middleware: (getDefaultMiddleware) =>
        getDefaultMiddleware({
          immutableCheck: false,
          serializableCheck: false,
        }),
      preloadedState: initialState,
      reducer: combined,
    }),
    ...renderOptions
  }: RenderWithReduxOptions = {}
): RenderResult => {
  const Wrapper: FC<PropsWithChildren> = ({ children }) => (
    <Provider store={store}>{children}</Provider>
  )

  return render(ui, { wrapper: Wrapper, ...renderOptions })
}
