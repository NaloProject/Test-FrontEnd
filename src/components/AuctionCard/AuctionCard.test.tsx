import AuctionCard from '@Components/AuctionCard/AuctionCard'
// import { screen } from '@Utils/reactTestingLibrary'
import { renderWithRedux } from '@Utils/reduxRenderer'
import { auctionCardStub } from './AuctionCard.stub'

describe('AuctionCard', () => {
  it('should mount first', async () => {
    const { container } = renderWithRedux(
      <AuctionCard auction={auctionCardStub} />
    )

    expect(container).toMatchSnapshot()
  })
})
