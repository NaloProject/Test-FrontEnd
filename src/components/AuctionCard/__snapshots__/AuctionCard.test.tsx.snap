// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuctionCard should mount first 1`] = `
<div>
  <article
    class="flex flex-col rounded-xl mr-1 sm:mr-2 md:mr-4 h-full"
  >
    <div
      class="flex flex-col w-full p-[16px] border border-slate-200 dark:border-slate-600 rounded-tl-xl rounded-tr-xl min-h-[400px]"
    >
      <header
        class="w-full relative mb-4"
      >
        <div
          class="overflow-hidden flex rounded-lg w-full relative w-full h-full !relative"
        >
          <a
            class="w-full h-full !relative"
            href="/auction/6"
          >
            <img
              alt="Ceres Searching"
              class="pointer-events-none object-cover !h-auto !relative"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=%2Fimages%2Fntfs%2FContemporary-Sculptures-For-The-Home.jpg&w=3840&q=75"
              srcset="/_next/image?url=%2Fimages%2Fntfs%2FContemporary-Sculptures-For-The-Home.jpg&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fntfs%2FContemporary-Sculptures-For-The-Home.jpg&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fntfs%2FContemporary-Sculptures-For-The-Home.jpg&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fntfs%2FContemporary-Sculptures-For-The-Home.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fntfs%2FContemporary-Sculptures-For-The-Home.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fntfs%2FContemporary-Sculptures-For-The-Home.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fntfs%2FContemporary-Sculptures-For-The-Home.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fntfs%2FContemporary-Sculptures-For-The-Home.jpg&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </a>
        </div>
        <div
          class="absolute bg-white bottom-[20px] flex left-[20px] rounded-lg px-3 py-2"
        >
          <span
            class="text-slate-700 font-medium mr-2 text-[12px]"
          >
            2 days
          </span>
          <span
            class="inline-block w-[12px] h-[12px]"
          >
            <svg />
          </span>
        </div>
      </header>
      <div
        class="flex mb-3"
      >
        <div
          class="flex-1"
        >
          <h3
            class="font-medium text-slate-700 dark:text-slate-300 mb-1"
          >
            Ceres Searching
          </h3>
          <h4
            class="text-[12px] text-slate-300 dark:text-slate-600"
          >
            2
             Editions minted
          </h4>
        </div>
        <div
          class="text-[13px] text-green-400"
        >
          1
           for sale
        </div>
      </div>
      <article
        class="flex items-center mt-auto"
      >
        <div
          class="relative"
        >
          <div
            class="overflow-hidden rounded-full h-[32px] w-[32px] flex-0 mr-4"
          >
            <img
              alt="Richard Noga"
              data-nimg="1"
              decoding="async"
              height="32"
              loading="lazy"
              src="/_next/image?url=%2Fimages%2FbestSellers%2FRichard-Noga.jpg&w=64&q=75"
              srcset="/_next/image?url=%2Fimages%2FbestSellers%2FRichard-Noga.jpg&w=32&q=75 1x, /_next/image?url=%2Fimages%2FbestSellers%2FRichard-Noga.jpg&w=64&q=75 2x"
              style="color: transparent;"
              width="32"
            />
          </div>
        </div>
        <div
          class="flex-1"
        >
          <h2
            class="font-bold text-[14px] text-slate-700 dark:text-slate-200 my-0 mt-auto"
          >
            Richard Noga
          </h2>
        </div>
      </article>
    </div>
    <footer
      class="flex p-[16px] border border-slate-200 dark:border-slate-600 border-t-0 min-h-[77px] items-center justify-between rounded-bl-xl rounded-br-xl w-full"
    >
      <div
        class="flex flex-col"
      >
        <span
          class="font-bold text-[14px] text-slate-700 dark:text-slate-300 mb-1"
        >
          10.89
           ETH
        </span>
        <span
          class="text-[13px] text-slate-400"
        >
          Starting Bid
        </span>
      </div>
      <div
        class="flex items-center"
      >
        <span
          class="font-medium text-[11px] text-slate-400 mr-1"
        >
          400
        </span>
        <span
          class="inline-block w-[22px] h-[22px]"
        >
          <svg
            class="fill-slate-400"
          />
        </span>
      </div>
    </footer>
  </article>
</div>
`;
